src/calendar/day/index.tsx

```diff
@@ -16,8 +16,9 @@ import PeriodDay from './period';
 function areEqual(prevProps: DayProps, nextProps: DayProps) {
   const prevPropsWithoutMarkDates = omit(prevProps, 'marking');
   const nextPropsWithoutMarkDates = omit(nextProps, 'marking');
+  const arePropsWithoutMarkedDatesEqual = isEqual(prevPropsWithoutMarkDates, nextPropsWithoutMarkDates);
```

Comment:
Is `isEqual` runs a deep check? 
If so, this makes the whole separation redundant. 
If you check for `isEqual` for both `markedDates` and the rest of the props you can just check the whole props object. 

The whole idea was to avoid deep equal check on the "other" props. cause deep check is considered heavy
But now I see that lodash's `omit` util creates a new object so it will return false when shallow comparing. 
Do we have a util for shallow comparison (one that runs on 1st level)?

Reply:
Yes, `isEqual` runs deep. The problem is when you check `prevPropsWithoutMarkDates === nextPropsWithoutMarkDates` it's always false (as the references are not equal...).
The purpose of the separation is to know if some other prop was updated, and if not deep check the `marking` prop if it was really changed before updating the component. This way it re-renders only the dates where the `marking` value changed

Reply:
Ok. I see what you mean... I can just us `isEqual` on all props.

Reply:
Ok sababa
`20.72 ms` with the fix? 
How much was it before?

Reply:
7.44 we need a shallow compare for the props

Reply:
ohh ok. 
(It's really nice that we can see the immediate impact.. )


Anyway, we can just iterate the otherProps and check them using `===` comparison, basically it'll check their reference has not changed and that's enough (that's what React does). 
We can use `_.find` or `_.some` to iterate all the props keys and compare